keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -149,22 +149,17 @@ metadata:
   finalizers:
     - cosi.io/finalizer [2]
 spec:
-  protocol:
-    name: [3]
-    version: [4]
-  bucketPrefix: [5]
-  bucketClassName: [6]
-  bucketInstanceName: [7]
+  bucketPrefix: [3]
+  bucketClassName: [4]
+  bucketInstanceName: [5]
```

Comment:
I'd like to think about the handshake for allocation. PV/PVC is pretty complicated and this doesn't have the same requirements (and therefore problems). I presume that, in the case of greenfield, some provisioner controller is going to write back to this object and fill in the bucketinstanceName ?

Would it make more sense to use status for "the instance you got" ? Is there any risk in letting users specify this arbitrarily?

Reply:
> in the case of greenfield, some provisioner controller is going to write back to this object and fill in the bucketinstanceName 

Yes. The central COSI controller will update BR.bucketInstanceName once the Bucket instance is created.

> Would it make more sense to use status for "the instance you got" ?

That would be consistent with the definition of status as representing the state of the world. However, there are APIs that define current state in `spec`, so not sure if strictly adhering to the API model is better than following existing patterns. Personally, I think it is probably better to follow the philosophy behind the APIs and define "current state" fields in status.

> Is there any risk in letting users specify this arbitrarily

This is where it seems less clear to me. We don't want users defining status fields. For brownfield, we do want users to reference the existing `Bucket` instance in their BR, and for greenfield, COSI filling in this field is a pseudo binding indicator. So, it seems BR.bucketInstanceName needs to be in spec, which is why today it's not in `status`.

Reply:
We have a mish-mash of patterns, so I apologize.  What I find awkward is that the COSI controller will write back to `spec`, which means it has RBAC access to do that.  I don't like controllers having access to `spec`.  Service PROBABLY should have used spec as a request and status as a confirmation of grant.

IOW, you could say `spec.bucketName: foo` is a request to bind to a Bucket "foo"; `status.bucketName: foo` indicates that is was actually bound.

I am going to write about this for sig-arch.